crates/net/network/src/peers/manager.rs

```diff
@@ -245,8 +254,8 @@ impl PeersManager {
     /// be scheduled.
     pub(crate) fn on_incoming_session_established(&mut self, peer_id: PeerId, addr: SocketAddr) {
         // we only need to check the peer id here as the ip address will have been checked at
-        // on_inbound_pending_session
-        if self.ban_list.is_banned_peer(&peer_id) {
+        // on_inbound_pending_session. We also check if the peer is in the backoff list here.
+        if self.ban_list.is_banned_peer(&peer_id) || self.backoff_list.contains_key(&peer_id) {
```

Comment:
hmm, I think we only backoff if we failed to establish a connection, so rejecting an incoming connection because we backed off is not reasonable imo

I think we can just neglect this

Reply:
No I actually agree but somehow the `test_shutdown` integration depends on this behaviour. I can rewrite the test as well?

Reply:
So the `test_shutdown` final assert relies on a `DisconnectRequested`, seeming to assume that its coming from the peer that actually shut down (P1). But that `DisconnectRequested` actually comes from the peer it's trying to connect to (P2), because P1 was banned (backed off) and therefore the connection was killed. I guess what should really happen when trying to connect to a peer after shutdown is it should just be ignored right?